spec/classes/collectd_plugin_apache_instance_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'spec_helper'
+
+describe 'collectd::plugin::apache::instance', type: :define do
+  on_supported_os(baseline_os_hash).each do |os, facts|
+    context "on #{os} " do
+      let :facts do
+        facts
+      end
+
+      options = os_specific_options(facts)
+      context 'simple usage' do
+        let :title do
+          'site2'
+        end
+        let :params do
+          {
+            'url': 'https://another.example.com',
+            'user': 'nobody',
+            'password': 'secrets',
+            'verifypeer': false,
+            'verifyhost': true,
+            'cacert': '/etc/foobar/ca.crt',
+            'sslciphers': 'TLS_CHACHA20_POLY1305_SHA256:TLS_AES_128_GCM_SHA256',
+            'timeout': 120,
+          }
+        end
+        it 'should create an instance file' do
+          content_instance_file = <<EOS
+<Plugin "apache">
+  <Instance "site2">
+    URL "https://another.example.com"
+    User "nobody"
+    Password "secrets"
+    VerifyPeer false
+    VerifyHost true
+    CACert "/etc/foobar/ca.crt"
+    SSLCiphers "TLS_CHACHA20_POLY1305_SHA256:TLS_AES_128_GCM_SHA256"
+    Timeout 120
+  </Instance>
+</Plugin>
+EOS
+          is_expected.to contain_file('apache-instance-site2.conf').with(
+            content: content_instance_file,
+            path: "#{options[:plugin_conf_dir]}/25-apache-instance-site2.conf"
+          )
+        end
```

Comment:
could you add another test to verify that it compiles with all dependencies?

Reply:
I have added check that required class are included, is that enough ?

Reply:
please add a `is_expected.to compile.with_all_deps`

Reply:
gotcha, done